src/compiler/crystal/semantic/call_error.cr

```diff
@@ -100,6 +100,10 @@ class Crystal::Call
           end
         end
 
+        if def_name == "allocate" && owner.is_a?(ModuleType) && owner.is_a?(MetaclassType)
```

Comment:
This condition should check if onwer is a MetaclassType and owner.instance_type.module?

The current check reads well but makes no sense because a MetaclassType inherits from ModuleType. It just happens to work because the only metaclasses that don't have an allocate are modules, though I think this check will also trigger for enums